src/GitHub.Api/Git/GitClient.cs

```diff
@@ -208,6 +208,15 @@ public interface IGitClient
         /// <returns>String output of git command</returns>
         ITask<string> DiscardAll(IOutputProcessor<string> processor = null);
 
+        /// <summary>
+        /// Executes at least one `git checkout` command to checkout files at the given changeset
+        /// </summary>
+        /// <param name="changeset">The md5 of the changeset</param>
+        /// <param name="files">The files to check out</param>
+        /// <param name="processor">A custom output processor instance</param>
+        /// <returns>String output of git command</returns>
+        ITask<string> CheckoutVersion(string changeset, IEnumerable<string> files, IOutputProcessor<string> processor = null);
```

Comment:
For the sake of immediate consistency we should probably take `IList<string>` here.
That being said, I'm not sure why we use `IList<string>`. I'll log a TODO issue for this.